### PR TITLE
dq-ReviewBackend get by user endpoint + tests

### LINKED ISSUE #12: Dining Commons Reviews Backend - Controller method for a user to get all of their reviews plus tests
format: GET /api/reviews?userId=x 

Gets all reviews from the database table that a single user created.

### PR DESCRIPTION
**Pull request by st4lemon**
Closes #12 

Go to https://dining-st4lemon.dokku-13.cs.ucsb.edu 

- On Swagger, verify that a get endpoint exists for userId
dininggetuser
- Get all reviews with /api/reviews/all, check what reviewerIds exist in the reviews
- Pick a reviewerId x, check that /api/reviews?userId=x returns all reviews by that user

- [ ] Logged out user cannot get by user id
- [ ] Logged in user/admin user can get reviews by user id
- [ ] All reviews made by the given user are returned 
- [ ] Only reviews made by the given user are returned

### CODE REVIEW

**Comment by clevermonkey16** [comment]
Seems like the dokku database might not be working
Screenshot 2024-11-23 at 11 52 32 AM

**Comment by st4lemon** [comment]
> Seems like the dokku database might not be working Screenshot 2024-11-23 at 11 52 32 AM

seems like postgres database was broken for some reason. issue should be fixed, I set up a new database and tested it again.

**Review by clevermonkey16 — APPROVED** [review]
LGTM, works on dokku

File: src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+     * @param dateServed        date item was served
+     * @param stars             rating from 0-5 inclusive
+     * @param reviewText        reviewer comments
+     * @param status            awaiting approval, approved, or rejected
+     * @param modId             user id of the moderator
+     * @param modComments       moderator comments about review
+     * @param createdDate       the date review was created
+     * @param lastEditedDate    the date review was last edited
+     * @return the saved review
+     */
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/dining/controllers/ReviewsController.java:84]: See Code Review feedback on #46 and apply it to the post method below as well

**Review by pconrad — CHANGES_REQUESTED** [review]
The two new methods LGTM, but this needs to be adjusted for the Code Review feedback on #46

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**